plan/plan.go

```diff
@@ -271,6 +271,16 @@ type Rel interface {
 	// result should be 3 columns consisting of the 5th, 2nd and 1st
 	// output columns from the underlying relation.
 	OutputMapping() []int32
+	// ClearMapping resets the mapping for this relation.
+	ClearMapping()
+	// ChangeMapping modifies the current relation by applying the provided
+	// mapping to the current relation.  Typically used to remove any unneeded
+	// columns or provide them in a different order.  If there already is a
+	// mapping on this relation, this provides mapping over the current mapping.
+	//
+	// If any column numbers specified are outside the currently available input
+	// range an error is returned and the mapping is left unchanged.
+	ChangeMapping(mapping []int32) error
```

Comment:
Can we have this make a copy as opposed to an inplace edit? 

I'm also more OverlayMapping or ApplyMapping as a name.

Also, would be good to use vararg.

Reply:
Renamed to `Remap()`.  There could be some confusion with `remap()` which we could rename to `applyMapping()`.